File: DuDoong-Domain/src/main/java/band/gosrock/domain/domains/comment/service/CommentDomainService.java
package band.gosrock.domain.domains.comment.service;


import band.gosrock.common.annotation.DomainService;
import band.gosrock.domain.domains.comment.domain.Comment;
import lombok.RequiredArgsConstructor;
import org.springframework.transaction.annotation.Transactional;

@DomainService
@RequiredArgsConstructor
public class CommentDomainService {

    @Transactional
    public void deleteComment(Comment comment, Long eventId) {
        comment.checkEvent(eventId);
        comment.delete();
    }
}

File: DuDoong-Domain/src/main/kotlin/band/gosrock/domain/domains/comment/service/CommentDomainService.kt
package band.gosrock.domain.domains.comment.service

import band.gosrock.common.annotation.DomainService
import band.gosrock.domain.domains.comment.domain.Comment
import org.springframework.transaction.annotation.Transactional

@DomainService
class CommentDomainService {

    @Transactional
    fun deleteComment(comment: Comment, eventId: Long) {
        comment.checkEvent(eventId)
        comment.delete()
    }
}
